src/components/dar_collection_table/CollectionConfirmationModal.js

```diff
@@ -0,0 +1,57 @@
+import {isNil} from "lodash/fp";
+import {h} from "react-hyperscript-helpers";
+import ConfirmationModal from "../modals/ConfirmationModal";
+import {isCollectionCanceled} from "../../libs/utils";
+import {getProjectTitle} from "./DarCollectionTable";
+
+export default function CollectionConfirmationModal(props) {
+  const {collection, showConfirmation, setShowConfirmation, cancelCollection, resubmitCollection} = props;
+
+  const getModalHeader = () => {
+    if(!isNil(collection)) {
+      return `${collection.darCode} - ${getProjectTitle(collection)}`;
+    }
+    return '';
+  };
+
+  const cancelOnClick = async() => {
+    await cancelCollection(collection);
+    setShowConfirmation(false);
+  };
+
+  const resubmitOnClick = async() => {
+    await resubmitCollection(collection);
+    setShowConfirmation(false);
+  };
+
+  const cancelModal = () =>
+    h(ConfirmationModal, {
+      showConfirmation,
+      styleOverride: {height: '35%'},
+      closeConfirmation: () => setShowConfirmation(false),
+      title: 'Cancel DAR Collection',
+      message: `Are you sure you want to cancel ${collection.darCode}?`,
+      header: getModalHeader,
+      onConfirm: cancelOnClick
+    });
+
+  const resubmitModal = () =>
+    h(ConfirmationModal, {
+      showConfirmation,
+      styleOverride: {height: '35%'},
+      closeConfirmation: () => setShowConfirmation(false),
+      title: 'Resubmit DAR Collection',
+      message: `Are you sure you want to resubmit ${collection.darCode}?`,
+      header: getModalHeader,
+      onConfirm: resubmitOnClick
+    });
+
+
+  if(isCollectionCanceled(collection) === true) {
```

Comment:
The browser's console log is throwing this error below
<img width="1788" alt="Screen Shot 2022-01-28 at 11 56 26 AM" src="https://user-images.githubusercontent.com/6664778/151589011-7a68bb8a-263e-4eb4-af1e-061c4526d388.png">

It's a bit esoteric, so I'll explain it a bit better. React Functional Components always return a template that is rendered whether or not there's data present or not. What you're doing here is returning the function, not the function result that produces a template.

To remove the error, you can just return the values of the function call (so `resubmitModal() instead of resubmitModal`). To make it even more react like, you can use ternary operators to define the output and return that instead.

`return isCollectionCanceled(collection) === true ? resubmitModal() : cancelModal()`